ee/server/service/software_installers.go

```diff
@@ -2239,6 +2274,67 @@ func packageExtensionToPlatform(ext string) string {
 	return requiredPlatform
 }
 
+func UpgradeCodeMigration(
```

Comment:
Could an optimization here be to keep a map of updated storage IDs to upgrade codes? That way we can avoid re-fetching and re-parsing files.

Or perhaps changing `UpdateInstallerUpgradeCode` to use the storage ID as the key, and that way all installers across teams with the same storage ID get updated at once?

Could help out in cases with lots of teams that all have overlapping software

Reply:
That's fair. I was patterning off of the uninstall scripts migration job but we can optimize a bit from there for cases where a customer has a bunch of teams all with the same package bytes on disk.

Reply:
Going to use the storage ID map idea rather than changing the update query. Reason being, if I change the update condition to storage ID + extension + upgrade code there will be a full table scan on all installers to grab the correct ones, as opposed to using a PK to update items one by one. Still makes sense to only download the same installer/calculate the upgrade code once though.